homeassistant/helpers/template.py

```diff
@@ -1496,6 +1496,15 @@ def forgiving_as_timestamp(value, default=_SENTINEL):
         return default
 
 
+def as_datetime(value, from_timestamp=False):
```

Comment:
I don't think we need to have this as a parameter (as this make use as a filter also harder).

Instead, we could check if we could parse it, if not, try to see if this is a UNIX timestamp instead.